### ISSUE TITLE
[feature request] start external perimeters inside objects

### ISSUE THREAD

**Issue by curiouspl2** [alexrj/Slic3r#2162]
to avoid blobs and pimples and to allow melt chamber pressure buildup,
slic3r could start external perimeters inside object for certain lenght (if there is any internal perimeter available ofcourse) 
and then just move along to external perimeter and form usual loop. 

i guess as quick hack it could be implemented as option to print most border internal perimeter before printing external perimeter. (which would result in printing those two first, if 'external perimeters first' is checked) .
note that with this option external perimeter would not really 'start' inside object, just the internal perimeter being printed first, and start/stop points overlapping would result in lack of drop in chamber pressure when moving from internal to external and lack of related 'startup blob'.
 

as bit more complex implementation both start and end of external perimeter could end inside object to prevent both blobbing and stringing being visible,  and a bit of combing (moving head above existing extrusion) could be used to prevent stringing even more (to 'wipe' head) 
this implementation would be much more 'clean' as all artifacts would be hidden inside object and there would be better integration with 'seam position' options.

**Comment by lordofhyphens** [comment]
Added to the open PR milestone.

**Comment by mrvn** [comment]
I thought this feature was rejected due to patent infringement?

**Comment by lordofhyphens** [comment]
@mrvn right, so it is. Things like this get called up a lot; I'll close this (missed the significance of the reference of #2243)

**Comment by Lenbok** [comment]
Stop scaremongering. #2243 doesn't provide any reference to the alleged patent, so how can you determine whether this issue or #2243 actually infringes? This issue should not be closed until such time.

**Comment by lordofhyphens** [comment]
@Lenbok Fine. I've got no interest in working on this myself.

**Comment by lordofhyphens** [comment]
@Lenbok added the referenced patent in #2243

**Comment by mrvn** [comment]
The relevant question is: Are you scared of the patent and won't accept a pull request for this?

**Comment by lordofhyphens** [comment]
@mrvn Depends on how much it resembles said patent. While @alexrj isn't in the US, I am and Slic3r is used in enough places that it would at least turn into a C&D from Stratysys if it gets pushed by them.

While I would review the code, this is something that I would happily leave to @alexrj to make the call on merging. If it came down to me, expect a higher burden of proof on the submitter that it actually does something worthwhile and it isn't going to cause issues.

I know there's an open thread re: that patent on stack exchange's ask patents site as well -- patents.stackexchange.com/questions/4667/3d-printing-seam-concealment-for-3d-models-patent-application-prior-art-re

**Comment by lordofhyphens** [comment]
regarding the thread, apparently Adrian Bowyer has the commit that's referenced as "prior art" ( @RRP-support ).

**Comment by RRP-support** [comment]
This is an automated reply to your message "Re: [alexrj/Slic3r] [feature request] start external perimeters
 inside objects (#2162)" sent to support@reprappro.com.

Dear Joseph Lenox <notifications@github.com>

This is an automated reply to your message "Re: [alexrj/Slic3r] [feature request] start external perimeters
 inside objects (#2162)" sent to support@reprappro.com.

Dear Joseph Lenox <notifications@github.com>

RepRap Professional Ltd. is now closed.

Note that our closure does not affect RepRapPro China Ltd (http://www.reprappro.com.cn/). That is a completely separate company (though we have obviously worked closely with them) and they will continue.

The market for low-cost 3D printers is now so crowded and so competitive that a small specialist company like ours could not expand. So, because we were not bankrupt and we did not have any debts to pay, we chose to stop while we were ahead and to concentrate on other activities.

We are an open-source company. This means that all our designs, software source-code and documentation are freely available online on our website and on the Github link below. All that information will continue to be available into the future.

Here is a list of places to look if you want to get parts for RepRapPro's designs:

The RepRap.org Website (http://reprap.org/wiki/RepRap_Buyers'_Guide)

eMaker (http://www.emakershop.com/Seller=7244)

RepRap Ltd (http://reprapltd.com/)

RepRapPro China Ltd (http://www.reprappro.com.cn/)

Think3DPrint3D (https://www.think3dprint3d.com/)

Trapletshop (http://gb.trapletshop.com/fisher-delta-3d-printer)


Replikeo (http://www.replikeo.com/en/)

RS Components (http://uk.rs-online.com/web/)

Mouser (http://www.mouser.co.uk/)

Farnell (http://uk.farnell.com/)

Toby (http://www.toby.co.uk/)




Our designs (including those for all the 3D-printed parts) are all available here: Github

If you donâ��t have access to a 3D Printer to make the parts for our designs, try 3Dhubs or your local hackerspace or makerspace. The latter two will also almost certainly have a lasercutter that will make the sheet parts for our designs as well.

We would like to thank everyone who has supported us in the past, either as customers or as contributors who have improved our open-source products, or both. The fact that, unusually for a former company, we will be able to leave megabytes of useful information available to everyone in perpetuity is due in part to the support from those people.

Other support channels are available. Please refer to the relevant troubleshooting guide for your machine:

Ormerod 2: https://reprappro.com/documentation/ormerod-2/troubleshooting/
Ormerod: https://reprappro.com/documentation/ormerod/troubleshooting/

Huxley Duo: https://reprappro.com/documentation/huxley-duo/troubleshooting/
Huxley: https://reprappro.com/documentation/huxley/troubleshooting/

Mono and Tricolour Mendel: https://reprappro.com/documentation/mendel-tricolour/troubleshooting/

Support is also available from other owners of RepRapPro and other 3D printers, either through the RepRap Forums: https://forums.reprap.org or through IRC which can be accessed from here: https://reprappro.com/support-channel/ (works on Chrome, not Firefox).

Best regards,

RepRapPro Support Team.

**Comment by lordofhyphens** [comment]
(quick test, sorry for spamming; need to make sure that autoresponder doesn't chime in again)

**Comment by lordofhyphens** [comment]
#1037 also seems relevant here. @nophead were you ever able to contact Adrian?

**Comment by nophead** [comment]
I did email him 24/05/2013 but he didn't reply and I forgot to chase it up.

**Comment by nophead** [comment]
http://blog.reprap.org/2009/07/build-quality.html

**Comment by VanessaE** [comment]
Looking forward to seeing this implemented, since Adrian's description clearly predates any of the aforementioned patents.
